ranger-gcs-permission-check-service/src/main/java/com/google/cloud/hadoop/ranger/gcs/permissionCheckService/RangerGcsPluginSingletonWrapper.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ *  * Copyright 2020 Google Inc. All Rights Reserved.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *    http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.google.cloud.hadoop.ranger.gcs.permissionCheckService;
+
+import com.google.cloud.hadoop.ranger.gcs.authorization.RangerGcsAuthorizer;
+import com.google.cloud.hadoop.ranger.gcs.permissionCheckService.utilities.ClusterRangerSettingChecker;
+import com.google.cloud.hadoop.ranger.gcs.utilities.RangerGcsPermissionCheckResult;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.FileNotFoundException;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Maintain a singleton Ranger plugin instance.
+ * Ranger plugin set it's policy engine during policy synchronization with Ranger server.
+ * Therefore, I need to maintain a singleton instance.
+ */
+public class RangerGcsPluginSingletonWrapper {
+    private static Log LOG = LogFactory.getLog(RangerGcsPluginSingletonWrapper.class);
+    private static RangerGcsPluginSingletonWrapper instance;
+
+    private RangerGcsAuthorizer authorizer = null;
+
+    private RangerGcsPluginSingletonWrapper () {
+        if (LOG.isDebugEnabled())
+            LOG.debug("==> RangerGcsPluginSingletonWrapper()");
+
+        authorizer = new RangerGcsAuthorizer();
+
+        if (LOG.isDebugEnabled())
+            LOG.debug("<== RangerGcsPluginSingletonWrapper()");
+    };
+
+    public static RangerGcsPluginSingletonWrapper getInstance() {
+        if (instance == null) {
+            synchronized (RangerGcsPluginSingletonWrapper.class) {
+                if (instance == null)
+                    instance = new RangerGcsPluginSingletonWrapper();
+            }
+        }
+
+        return instance;
+    }
+
+    public void init() {
+        if (authorizer != null)
+            authorizer.init();
+    }
+
+    public void stop() {
+        if (authorizer != null)
+            authorizer.stop();
+    }
+
+    public RangerGcsPermissionCheckResult isAccessAllowed(String user, Set<String> userGroups,
```

Comment:
Please rename this method to something like "produceRangerGcsPermissionCheckResult". Normally "isXXX" indicates the return type is Boolean.